integration-tests/src/test_loop/tests/resharding_v3.rs

```diff
@@ -387,6 +389,71 @@ fn next_block_has_new_shard_layout(epoch_manager: Arc<dyn EpochManagerAdapter>,
         && shard_layout != next_shard_layout
 }
 
+/// Asserts that for each child shard (Mem)Trie and FlatState contains the same final state.
+fn assert_state_sanity_for_children_shard(parent_shard_uid: ShardUId, client: &Client) {
+    let final_head = client.chain.final_head().unwrap();
+
+    for child_shard_uid in client
+        .epoch_manager
+        .get_shard_layout(&final_head.epoch_id)
+        .unwrap()
+        .get_children_shards_uids(parent_shard_uid.shard_id())
+        .unwrap()
+    {
+        let state_root = *client
+            .chain
+            .get_chunk_extra(&final_head.prev_block_hash, &child_shard_uid)
```

Comment:
Should we be getting the state root from chunk header instead of chunk extra? Or is this because we are looking for post state root?

Reply:
Correct, we are looking for the post state root